### DISCUSSION TITLE
From FIBO OWL to FIBO-V

### DISCUSSION THREAD

**Discussion by trypuz** [edmcouncil/ontology-publisher#40] [FIBO Vocabulary]
@mereolog, please share with us how you see a transition from FIBO OWL to FIBO-V so that we can comment on it and agree on its final shape.

Cc: @kptyson, @dallemang, @rivettp, @sklarman

See also: https://github.com/edmcouncil/ontology-publisher/issues?q=is%3Aissue+is%3Aopen+label%3A%22FIBO+vocabulary%22

**Comment by mereolog** [comment]
Here is my vision on how it might happen given @trypuz's point made in this [post](https://github.com/edmcouncil/ontology-publisher/issues/35#issuecomment-804043210).
1. FIBO class:
a. gets translated to Vocabulary as an instance of skos:Concept:
b. the concept gets a new IRI by replacing 'ontology' infix in the class' IRI by 'vocabulary'
c. all its annotation properties are translated as such to the vocabulary's concept, except for the two: rdfs:label is translated as skos:prefLabel and av:synonym as skos:altLabel.
d. in addition, the concept: 
(i) gets a value for skos:inScheme property, e.g., fibo-skos:ClassConcepts
(ii) gets a pointer to its OWL class (via rdfs:isDefinedBy)
e. all its sub/superclasses are connected to the concept by skos:broader
f. all its OWL restrictions are dropped
2. FIBO object property
a. gets translated as an object property to Vocabulary
b. all its annotation properties are translated in the same way as owl classes
c. rdfs:domain and rdfs:range are translated to the respective Vocabulary concepts
d. all its sub/superproperties are translated as such, i.e., via owl:subPropertyOf
e. all its logical characteristics is dropped, e.g, owl:inverseOf, owl:Reflexive, etc.
3. FIBO annotation property:
a. gets translated as an annotation property to Vocabulary
4. FIBO data property:
a. is dropped, i.e., is not translated
5. FIBO named individual:
a. is dropped
6. FIBO blank node:
a. is dropped
7. FIBO ontology:
a. is dropped - this implies that all owl imports triples are dropped as well.

I know that the current code is slightly different from the spec above, but I tried to "rationalize" it a little.

**Reply by trypuz** [reply]
@kptyson, @dallemang, @rivettp, @sklarman, any comments?

**Reply by rivettp** [reply]
Not sure of the purpose of this discussion given that @kptyson made a very explicit code proposal in the Issue you linked to. Which I largely agree with.
Nonetheless here are a few comments
1.c: I suggest retaining rdfs:label in addition to skos:prefLabel for compatibility
1.d(i): I don't think ClassConcepts is useful as a Scheme. Why not transform the original Ontology or maybe the Module into a Scheme (see 7). Instead I suggest we have top level Concepts for ClassConcept and PropertyConcept and IndividualConcept to indicate what type of thing the concept was in OWL.
2. This generally seems a lot less clear than the description for Class. Since SKOS does not have "object property" what is the effect of 1.a? FWIW  thought we were going to make these Concepts too as per @kptyson's code. Linked to super properties via skos:broader. Since SKOS does not have subPropertyOf. 

3. Unclear since SKOS does not have annotation properties. FIBO has so few and most are based on SKOS anyway so I think each could be replaced in triples by the closest SKOS equivalent.
4. I strongly disagree with dropping data properties - why not treat them the same way as proposed for object properties?
5. Likewise I disagree with dropping individuals - they can represent important concepts e.g. the other day we were talking about AmericanStyle and EuropeanStyle for Options which are represented by NamedIndividuals.
7. Ontologies (or possibly Modules) should be retained as ConceptSchemes since they are useful ways to manage/organize sets of concepts. A single flat scheme will be overwhelmingly useless

**Reply by mereolog** [reply]
Given that @trypuz raised [this](https://github.com/edmcouncil/ontology-publisher/issues/35#issuecomment-804043210) in reply to @kptyson proposal, I think that @trypuz should address your, @rivettp, worries.
